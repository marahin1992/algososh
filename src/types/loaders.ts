export enum Loader {
  Add = 'add',
  Del = 'del',
}