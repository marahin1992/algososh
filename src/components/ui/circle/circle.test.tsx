import React from 'react';
import renderer from 'react-test-renderer';

import { Circle } from "./circle";
import { ElementStates } from '../../../types/element-states';

it('Кружок без буквы рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок с буквой рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю"/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок с head рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" head='Рис'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок с react-элементом в head рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" head={<Circle/>}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок с tail рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" tail='Рис'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок с react-элементом в tail рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" tail={<Circle/>}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок с index рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" index={7}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок с пропом isSmall ===  true рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" isSmall={true}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок в состоянии default рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" state={ElementStates.Default}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок в состоянии changing рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" state={ElementStates.Changing}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кружок в состоянии modified рендерится без ошибок', () => {
  const tree = renderer
    .create(<Circle letter="Ю" state={ElementStates.Modified}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

