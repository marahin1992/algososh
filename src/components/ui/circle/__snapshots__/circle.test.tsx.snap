// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кружок без буквы рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок в состоянии changing рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок в состоянии default рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок в состоянии modified рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    Рис
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с index рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    7
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Кружок с react-элементом в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с react-элементом в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Кружок с tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Рис
  </div>
</div>
`;

exports[`Кружок с буквой рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с пропом isSmall ===  true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Ю
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
